lib/minetools/server_status_tool/server_status.rb

```diff
@@ -10,24 +10,18 @@ def initialize options={host: nil, port: nil}
 				@port = options[:port]
 				@port = 25565 if @port.nil?   # default
 				@status = nil
+				@socket = nil
 
 				check_instance_variables! e=ArgumentError
 			end
 
+			def socket
+				@socket ||= TCPSocket.new(@host, @port, connect_timeout: 1)
+			end
```

Comment:
socketだと、socket単体を示してる感がある
```suggestion
			def new_socket
				return TCPSocket.new(@host, @port, connect_timeout: 1)
			end
```
もしくは、initialize_socketとかにするとか。
（好みの問題、するーしてもいい。）

Reply:
結果的にsocketインスタンスを返すので、socketでいきます。

状態を持ったsocketインスタンスを返す必要があるので、一度instance変数に格納してますが、mockの仕込み安さの観点でメソッドにし直しています。

初回実行時に`TCPSocket.new()`され、以後はinstance変数からすでにあるinstanceが返ります。(メモ化)
効果としては `@socket = @socket || TCPSocket.new()` と変わらないです。

`new_socket()` とかだと、2回目以降の実行のときの戻り値が新たに生成されるので意味が変わってしまいます...

(合意済み)